File: ProjectEuler9/main.cpp
#include <iostream>
#include <chrono>
#include <ctime>
#include <cmath>

int main() {
    std::chrono::time_point<std::chrono::system_clock> start, end;
    start = std::chrono::system_clock::now();

    int y = 1000;
    int a, b, c;
    bool solutionFound = false;

    for (a = 1; a < y/3; a++) {
        for (b = 1; b < y/2; b++) {
            double potentialC = sqrt(pow(a, 2) + pow(b, 2));
            if ((int) potentialC == potentialC) {
                c = (int) potentialC;
                if (a + b + c == y) {
                    solutionFound = true;
                    break;
                }
            }
        }
        if (solutionFound) break;
    }

    end = std::chrono::system_clock::now();
    long elapsed_millis = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
    std::cout << "Elapsed Milliseconds: " << elapsed_millis << std::endl;

    std::cout << "a: " << a << " b: " << b << " c: " << c << std::endl;
    std::cout << a * b * c << std::endl;
    return 0;
}

File: ProjectEuler9/src/main.rs
use std::time::Instant;

fn main() {
    let now = Instant::now();

    let y = 1000;
    let mut a = 0;
    let mut b = 0;
    let mut c= 0;

    'outer: for i in 1..=y/3 {
        for j in 1..=y/2  {
            let potential_c = (((i as i32).pow(2) + (j as i32).pow(2)) as f64).sqrt();

            if potential_c.floor() == potential_c {
                if i + j + (potential_c as i32) == y {
                    a = i as i32;
                    b = j as i32;
                    c = potential_c as i32;
                    break 'outer;
                }
            }
        }
    }

    println!("Product: {}", a*b*c);
    println!("Elapsed µs: {}", now.elapsed().as_micros());
}